tests from px_file_test.py fail because localhost != localhost.localdomain

when i run `./test.sh` i consistently get following errors

```
    def test_listen_name():
        file = lsof_to_file(["f6", "au", "tIPv4", "d0x42", "nlocalhost:63342"])
        assert file.name == "localhost:63342"
>       assert str(file) == "[IPv4] localhost:63342 (LISTEN)"
E       AssertionError: assert '[IPv4] local...3342 (LISTEN)' == '[IPv4] localh...3342 (LISTEN)'
E         - [IPv4] localhost.localdomain:63342 (LISTEN)
E         ?                 ------------
E         + [IPv4] localhost:63342 (LISTEN)

tests/px_file_test.py:81: AssertionError
```
https://github.com/walles/px/blob/b1731c7f3359bcfda5e3f4a33d53c23befd337cd/tests/px_file_test.py#L78

and
```
    def test_str_resolve():
        # FIXME: This will break if Google changes the name of 8.8.8.8
        test_me = px_file.PxFile()
        test_me.type = "IPv4"
        test_me.name = "127.0.0.1:51786->8.8.8.8:https"
>       assert str(test_me) == "[IPv4] localhost:51786->google-public-dns-a.google.com:https"
E       AssertionError: assert '[IPv4] local...gle.com:https' == '[IPv4] localh...gle.com:https'
E         - [IPv4] localhost.localdomain:51786->google-public-dns-a.google.com:https
E         ?                 ------------
E         + [IPv4] localhost:51786->google-public-dns-a.google.com:https

tests/px_file_test.py:149: AssertionError
```
https://github.com/walles/px/blob/b1731c7f3359bcfda5e3f4a33d53c23befd337cd/tests/px_file_test.py#L144

it seems that on my system `localhost` is `localhost.localdomain`.

similar to bug #30 this should be solved so that input data is not taken from the running system. and even more important not taken from the internet. input data should be provided as hardcoded known good literals. at least for unit tests. when doing acceptance tests input data can be taken from the running system but then arrangements should be done to handle the differences when asserting.

--- a/px/px_file.py
+++ b/px/px_file.py
@@ -125,12 +125,19 @@ def resolve_endpoint(endpoint):
         address = address[1:-1]
 
     port = endpoint[splitindex + 1:]
+    host = None
 
     try:
-        return socket.gethostbyaddr(address)[0] + ":" + port
+        host = socket.gethostbyaddr(address)[0]
     except Exception:
         # Lookup failed for whatever reason, give up
         return endpoint
+
+    if host is "localhost.localdomain":
+        # "localdomain" is just a long word that doesn't add any information
+        host = "localhost"
+
+    return host + ":" + port
 
 
 def call_lsof():

